oak_marketplace_emotion_recognition/main.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+import cv2
+import numpy as np
+import skvideo.io
+from emotion_recognition_retail import InferenceModel, process_frame
+from modelplace_api.visualization import draw_emotion_recognition_one_frame
+from args_parser import parse_args
+from utils import EmotionGraphsCollector, overlay_image_alpha, place_class_names_and_percents, X_OFFSET, Y_OFFSET
+
+
+
+
+def build_model(model_path):
+    model = InferenceModel(model_path=model_path)
+    model.model_load()
+    model.add_cam_to_pipeline()
+    return model
+
+
+def main(args):
+    model_path = os.path.join(args.root_model_path, "checkpoint")
+    model = build_model(model_path)
+    input_width, input_height = model.get_input_shapes()
+    emotions_collector = EmotionGraphsCollector()
+    proceed = True
+    visualization_results = []
+    overlay = cv2.imread("overlay_450x450.png", cv2.IMREAD_UNCHANGED)
+
+    while proceed:
+        image = np.ascontiguousarray(
+            model.get_frame_from_camera()
+                .reshape((3, input_height, input_width))
+                .transpose(1, 2, 0),
+        )
+        ret, proceed, _ = process_frame(image, model, visualization_func=None)
+        if ret:
+            class_name = ret[0].emotions[0].class_name
+            emotions_collector.update_graph(class_name)
+        image = cv2.resize(image, (450, 450))
+        vis_result = draw_emotion_recognition_one_frame(image, ret)
```

Comment:
You've resized the image but you didn't change so then you also must resize bboxes because it's required for visualization